amaranth/sim/pysim.py

```diff
@@ -229,17 +271,75 @@ def commit(self):
         return awoken_any
 
 
+class _PyMemoryState(BaseMemoryState):
+    __slots__ = ("memory", "data", "pending_writes", "waiters", "pending")
+
+    def __init__(self, memory, pending):
+        self.memory = memory
+        self.pending = pending
+        self.waiters = set()
+        self.reset()
+
+    def reset(self):
+        self.data = list(self.memory._init)
+        self.pending_writes = {}
```

Comment:
So the way CXXRTL solves the problem of order dependency is by maintaining a queue (`list` instead of a `dict`). In the interest of commonality, could you use a queue here too?

Reply:
The `write` implementation relies on this actually being a dict.